src/plugins/intel_cpu/src/config.cpp

```diff
@@ -132,58 +134,79 @@ void Config::readProperties(const std::map<std::string, std::string> &prop) {
         } else {
             IE_THROW(NotFound) << "Unsupported property " << key << " by CPU plugin";
         }
-        _config.clear();
     }
+
+    if (!prop.empty())
+        _config.clear();
+
     if (exclusiveAsyncRequests)  // Exclusive request feature disables the streams
         streamExecutorConfig._streams = 1;
 
     CPU_DEBUG_CAP_ENABLE(readDebugCapsProperties());
     updateProperties();
 }
+
 void Config::updateProperties() {
-    if (!_config.size()) {
-        switch (streamExecutorConfig._threadBindingType) {
-            case IStreamsExecutor::ThreadBindingType::NONE:
-                _config.insert({ PluginConfigParams::KEY_CPU_BIND_THREAD, PluginConfigParams::NO });
-            break;
-            case IStreamsExecutor::ThreadBindingType::CORES:
-                _config.insert({ PluginConfigParams::KEY_CPU_BIND_THREAD, PluginConfigParams::YES });
-            break;
-            case IStreamsExecutor::ThreadBindingType::NUMA:
-                _config.insert({ PluginConfigParams::KEY_CPU_BIND_THREAD, PluginConfigParams::NUMA });
-            break;
-            case IStreamsExecutor::ThreadBindingType::HYBRID_AWARE:
-                _config.insert({ PluginConfigParams::KEY_CPU_BIND_THREAD, PluginConfigParams::HYBRID_AWARE });
-            break;
-        }
-        if (collectPerfCounters == true)
-            _config.insert({ PluginConfigParams::KEY_PERF_COUNT, PluginConfigParams::YES });
-        else
-            _config.insert({ PluginConfigParams::KEY_PERF_COUNT, PluginConfigParams::NO });
-        if (exclusiveAsyncRequests == true)
-            _config.insert({ PluginConfigParams::KEY_EXCLUSIVE_ASYNC_REQUESTS, PluginConfigParams::YES });
-        else
-            _config.insert({ PluginConfigParams::KEY_EXCLUSIVE_ASYNC_REQUESTS, PluginConfigParams::NO });
-        if (enableDynamicBatch == true)
-            _config.insert({ PluginConfigParams::KEY_DYN_BATCH_ENABLED, PluginConfigParams::YES });
-        else
-            _config.insert({ PluginConfigParams::KEY_DYN_BATCH_ENABLED, PluginConfigParams::NO });
+    if (!_config.empty())
+        return;
+
+    auto toConfigString = [](const ov::Any& property) -> std::string {
+        std::stringstream strm;
+        property.print(strm);
+        return strm.str();
+    };
 
-        _config.insert({ PluginConfigParams::KEY_DYN_BATCH_LIMIT, std::to_string(batchLimit) });
-        _config.insert({ PluginConfigParams::KEY_CPU_THROUGHPUT_STREAMS, std::to_string(streamExecutorConfig._streams) });
-        _config.insert({ PluginConfigParams::KEY_CPU_THREADS_NUM, std::to_string(streamExecutorConfig._threads) });
-        IE_SUPPRESS_DEPRECATED_START
+    switch (streamExecutorConfig._threadBindingType) {
+    case IStreamsExecutor::ThreadBindingType::NONE:
+        _config.insert({ PluginConfigParams::KEY_CPU_BIND_THREAD, PluginConfigParams::NO });
+        _config.insert({ ov::affinity.name(), toConfigString(ov::Affinity::NONE) });
```

Comment:
The idea is that we store both new and legacy config string keys since we have to return both
to support both legacy and new scenarios